plugin-guide.md

```diff
@@ -9,10 +9,17 @@ using the `provides` keyword. All plugins need to depend only on the
 infrastructure.
 
 There are several services that a plugin can provide and a plugin is free to
-provide multiple. Each has two classes: a small "loader" class responsible for
-loading this plugin's configuration from JSON files and a "real" interface that
-does most of the work. For loading, each plugin must provide a unique name that
-will be used in the `"type"` property of JSON configuration files.
+provide multiple. Plugins are loaded from JSON data in the Vidarr configuration
+file or, in the case of an unload filter, user requests, using Jackson. Each
+plugin can load whatever Jackson-compatible data from JSON it requires. Each
+plugin has a small "provider" class which provides type information for
+Jackson. In the JSON file, the `"type"` attribute will be used to create the
```

Comment:
Example json (and maybe an example of the matching provider class type method) would be nice, though not the most pressing need since one could look at the config on the existing vidarr server